src/components/DateTimeField/index.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { InputText } from '@edx/paragon';
+import { getDateString, getTimeString } from '../../utils/index';
+import FieldLabel from '../FieldLabel';
+import { DATE_FORMAT } from '../../data/constants';
+
+class DateTimeField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      date: getDateString(this.props.input.value) || '',
+      time: getTimeString(this.props.input.value) || '',
+    };
+    this.concatDateTime = this.concatDateTime.bind(this);
+    this.updateDate = this.updateDate.bind(this);
+    this.updateTime = this.updateTime.bind(this);
+  }
+
+  concatDateTime(date, time) {
+    const datetime = moment.utc(date + time, 'YYYY/MM/DD HH:mm').format(DATE_FORMAT);
+    this.props.input.onChange(datetime);
+  }
+
+  updateDate(value) {
+    const {
+      time,
+    } = this.state;
+    this.setState({ date: value });
+    this.concatDateTime(value, time);
+  }
+
+  updateTime(value) {
+    const {
+      date,
+    } = this.state;
+    this.setState({ time: value });
+    this.concatDateTime(date, value);
+  }
+
+  render() {
+    const {
+      dateName,
+      dateLabel,
+      timeName,
+      timeLabel,
+      helpText,
+      courseInReview,
+    } = this.props;
+
+    const {
+      date,
+      time,
+    } = this.state;
+
+    return (
+      <div className="row">
+        <div className="col-6">
+          <InputText
+            name={dateName}
+            type="date"
+            value={getDateString(date)}
+            label={
+              <FieldLabel
+                id={`${dateName}-label`}
+                text={dateLabel}
+                required
```

Comment:
You dropped the (addmittedly extraneous) `requiredForSubmit` on these labels. No biggy, since they were excess. But just noting.

More importantly, it's not typical for a component to assume that it is required in all forms it's put into. That should probably be a prop.